example/src/main/java/org/apache/rocketmq/example/ordermessage/OrderedProducer.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.rocketmq.example.ordermessage;
+
+import java.util.List;
+import org.apache.rocketmq.client.producer.DefaultMQProducer;
+import org.apache.rocketmq.client.producer.MessageQueueSelector;
+import org.apache.rocketmq.client.producer.SendResult;
+import org.apache.rocketmq.common.message.Message;
+import org.apache.rocketmq.common.message.MessageQueue;
+import org.apache.rocketmq.remoting.common.RemotingHelper;
+
+public class OrderedProducer {
+
+    private static final int ORDER_NUM = 10;
+
+    public static void main(String[] args) throws Exception {
+
+        int[] orderCounts = new int[ORDER_NUM];
+
+        DefaultMQProducer producer = new DefaultMQProducer("please_rename_unique_group_name");
+
+        producer.start();
+
+        String[] tags = new String[] {"TagA", "TagB", "TagC", "TagD", "TagE"};
+        for (int i = 0; i < 100; i++) {
+            int orderId = i % ORDER_NUM;
+            Message msg =
+                new Message("OrderedMsgTest", tags[i % tags.length], "orderID" + orderId +
+                    ":" + orderCounts[orderId]++ + "th",
+                    ("Hello RocketMQ " + i).getBytes(RemotingHelper.DEFAULT_CHARSET));
+            /*
+             * RocketMQ provides ordered messages using FIFO order
+             * Messages with the same orderID are sent to the same message queue
+             */
+            SendResult sendResult = producer.send(msg, new MessageQueueSelector() {
+                @Override
+                public MessageQueue select(List<MessageQueue> mqs, Message msg, Object arg) {
+                    Integer id = (Integer) arg;
+                    int index = id % mqs.size();
```

Comment:
@RongtongJin Here need to check whether the mqs are as expected. 
For example, two broker named 'broker1' and 'broker2', if broker1 get crashed, then the mqs maybe changed, the hashed value is changed too.
So if user want strictly ordered message, he need to know how many brokers and how many queues to be hashed at the application side.